modules/lofreq/1.1/lofreq.smk

```diff
@@ -122,25 +124,50 @@ rule _lofreq_preprocess_normal_unmatched:
+        vcf_snvs_stringent = CFG["dirs"]["lofreq_somatic"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/normal_stringent.snvs.vcf.gz",
+        normal_relaxed_log = CFG["dirs"]["lofreq_somatic"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/normal_relaxed.log",
+        preprocessing_complete = CFG["dirs"]["lofreq_somatic"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/preprocessing.complete"
+    shell:
+        op.as_one_line("""
+        out_dir=$(dirname {output.vcf_relaxed});
+        mkdir -p $out_dir;
+        ln -f {input.vcf_relaxed} $out_dir;
+        ln -f {input.vcf_relaxed}.tbi $out_dir;
+        ln -f {input.vcf_indels_stringent} $out_dir;
+        ln -f {input.vcf_indels_stringent}.tbi $out_dir;
+        ln -f {input.vcf_snvs_stringent} $out_dir;
+        ln -f {input.vcf_snvs_stringent}.tbi $out_dir;
+        ln -f {input.normal_relaxed_log} $out_dir;
+        ln -f {input.preprocessing_complete} $out_dir;
+        """)
+
 # Run LoFreq in somatic variant calling mode for unpaired tumours using precomputed SNV positions from unmatched normal
 # For unmatched tumours we need to symlink all the files from the preprocessing to the pair's output directory
 # Matched and unmatched tumours are handled identically here. The normal is only run once and all unmatched tumours that rely on it
 # won't be run until that processing is done. 
 # The creation of symbolic links in this rule could probably be separated out into a rule that uses Oncopipe's relative_symlink
-rule _lofreq_run_tumour_unmatched:
+rule _lofreq_run_tumour:
     input:
-        vcf_relaxed = rules._lofreq_preprocess_normal_unmatched.output.vcf_relaxed,
-        vcf_indels_stringent = rules._lofreq_preprocess_normal_unmatched.output.vcf_indels_stringent,
-        vcf_snvs_stringent = rules._lofreq_preprocess_normal_unmatched.output.vcf_snvs_stringent,
-        normal_relaxed_log = rules._lofreq_preprocess_normal_unmatched.output.normal_relaxed_log,
-        preprocessing_complete = rules._lofreq_preprocess_normal_unmatched.output.preprocessing_complete,
+        preprocessing_complete = rules._lofreq_link_to_preprocessed.output.preprocessing_complete,
         tumour_bam = CFG["dirs"]["inputs"] + "bam/{seq_type}--{genome_build}/{tumour_id}.bam",
         normal_bam = CFG["dirs"]["inputs"] + "bam/{seq_type}--{genome_build}/{normal_id}.bam",
+        vcf_relaxed = rules._lofreq_link_to_preprocessed.output.vcf_relaxed,
         fasta = reference_files("genomes/{genome_build}/genome_fasta/genome.fa"),
         dbsnp = reference_files("genomes/{genome_build}/variation/dbsnp.common_all-151.vcf.gz"), #in our experience, this filter doesn't remove as many SNPs as one would expect
         bed = reference_files("genomes/{genome_build}/genome_fasta/main_chromosomes.bed")
```

Comment:
I think we could make this configurable to use exome beds with a single line of code like this at the top of the module Snakefile:  
```
CFG['switches']['regions_bed']['_default'] = reference_files("genomes/{genome_build}/genome_fasta/main_chromosomes.bed") 
```
Then in the lofreq_run rule(s): 

```
rule _lofreq_run: 
    input: 
        ...
        bed = op.switch_on_wildcard("seq_type", CFG["switches"]["regions_bed"])
```

If no bed files are specified under additional keys in the `regions_bed` part of the config, it will use the default reference_workflow chromosome bed.

Reply:
I had spotted a line of code that looked almost identical to this (and commented out) in one of the edited snakefiles Jasper was using. What would the corresponding section of the config look like for this to use other bed files?

Reply:
It currently looks like this:
```
regions_bed:
                _default: ""
                # Set to "" if you have no capture data
                capture: "-l __UPDATE__"
```
But you can't supply a second bed file argument to lofreq (it ignores the second one).

Reply:
In this update it would change only slightly to: 
```
regions_bed:
                _default: ""
                # Set to "" if you have no capture data
                capture: ""
```

Reply:
Should we completely remove this part? If I understand correctly, it doesn't matter what the user puts there it would get immediately clobbered anyway, right? Or does it need to be there so that it can be replaced later? 
 _default: ""

Reply:
Yes it could be left blank, but it makes the code to put the default bed file into the config more complicated because the key is missing. I think it would be reasonable to leave it in the default config with a comment telling the user what gets used there and that any custom BED files must be specified with a seq_type wildcard.